File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int i=0,j=0;
        while(i<n && j<n){
            if(nums[i]!=0 && nums[j]!=0){
                i++;
                j++;
            }
           else if(nums[i]==0 && nums[j]==0){
                j++;
            }
           else if(nums[i]==0 && nums[j]!=0){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                i++;
                j++;
            }
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        i,j=0,0
        while(j<n):
            if  nums[i]==0 and nums[j] !=0 :
                nums[i], nums[j] = nums[j], nums[i]
                i += 1
            elif nums[i]!=0 and nums[j]!=0:
                i += 1
            j += 1    
